File: backend/internal/entities/card.go
package entities

type Card struct {
	Id           int    `json:"-" db:"id"`
	Organization string `json:"organization" binding:"required" db:"organization"`
	Number       string `json:"number" db:"number"`
	Category
}

type Category struct {
	Id           int    `json:"-" db:"id"`
	CategoryName string `json:"category_name" db:"category_name"`
}

File: mobile/hand_card/lib/model/card.dart
import 'dart:ffi';

import 'package:flutter/material.dart';

class DiscountCard {
  final int id;
  final String organization;
  final String categoryName;
  final String number;

  const DiscountCard ({
    required this.id,
    required this.organization,
    required this.categoryName,
    required this.number,
  });

static DiscountCard fromJson(json) => DiscountCard(
  id: json['id'],
  organization: json['organization'],
  categoryName: json['category_name'],
  number: json['number']);
}
